### PR TITLE
Headless mode readiness probe

### PR DESCRIPTION
**Pull request by chatton**
### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

This PR updates the operator to configure the readiness probe in headless mode, instead of using just the health file as it was doing.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,7 +1,6 @@
 .github
 .idea
 agent
-build/_output
```
Comment (chatton) [inline .dockerignore:4]: allows us to use `operator-sdk build` for faster local development

File: README.md [hunk 2]
```diff
@@ -111,7 +111,7 @@ If you want to deploy the operator on OpenShift you will have to provide the env
 
 See [here](deploy/crds/mongodb.com_v1_mongodb_openshift_cr.yaml) for an example of how to provide the required configuration for a MongoDB ReplicaSet.
 
-See [here](deploy/operator_openshift.yaml) for an example of how to configure the Operator deployment.
+See [here](deploy/openshift/operator_openshift.yaml) for an example of how to configure the Operator deployment.
```
Comment (chatton) [inline README.md:114]: moved the openshift example into another directory so it is still possible to do. `kubectl apply -f deploy/` without deploying the operator twice

File: build/Dockerfile [hunk 3]
```diff
@@ -4,6 +4,14 @@ ENV OPERATOR=/usr/local/bin/mongodb-kubernetes-operator \
     USER_UID=1001 \
     USER_NAME=mongodb-kubernetes-operator
 
+RUN apt-get update  && \
+    apt-get install -y curl
+
+ENV manifest_version=4.4
+RUN mkdir -p /content/ \
+        && chmod -R +r /content/ \
+        && curl --fail --retry 3 -o /usr/local/version_manifest.json "https://opsmanager.mongodb.com/static/version_manifest/${manifest_version}.json"
```
Comment (chatton) [inline build/Dockerfile:13]: small update to add the version manifest to the "faster" local dev build using `operator-sdk build`

File: pkg/controller/mongodb/replica_set_controller.go [hunk 4]
```diff
@@ -609,6 +614,23 @@ func mongodbAgentContainer(volumeMounts []corev1.VolumeMount) container.Modifica
 		},
 		),
 		container.WithEnvs(
+			corev1.EnvVar{
+				Name:  headlessAgentEnv,
```
Comment (chatton) [inline pkg/controller/mongodb/replica_set_controller.go:618]: flag the readiness probe to go down the headless agent route

File: pkg/controller/mongodb/replica_set_controller.go [hunk 5]
```diff
@@ -609,6 +614,23 @@ func mongodbAgentContainer(volumeMounts []corev1.VolumeMount) container.Modifica
...
+					FieldRef: &corev1.ObjectFieldSelector{
+						APIVersion: "v1",
+						FieldPath:  "metadata.namespace",
+					},
+				},
+			},
+			corev1.EnvVar{
+				Name:  automationConfigEnv,
+				Value: automationConfigSecretName,
+			},
```
Comment (chatton) [inline pkg/controller/mongodb/replica_set_controller.go:633]: provide the readiness probe information about which secret to read from

**Review by rodrigovalin — APPROVED** [review]
LGTM

what do you think about renaming `status.members` to `status.currentMembers` or something similar that indicates that it is a transient value that moves towards `spec.members`... ? Like the `readyReplicas` from STS? (no need for this PR)

**Comment by chatton** [comment]
@rodrigovalin I like your suggestion about the renaming! I think we can do this separately like you said 👍

**Pull request merged**